apps/string_publisher/app.c

```diff
@@ -0,0 +1,82 @@
+#include <stdio.h>
+#include <unistd.h>
+
+#include <rcl/rcl.h>
+#include <rcl/error_handling.h>
+#include <std_msgs/msg/string.h>
+
+#include <rclc/rclc.h>
+#include <rclc/executor.h>
+
+#ifdef ESP_PLATFORM
+#include "freertos/FreeRTOS.h"
+#include "freertos/task.h"
+#endif
+
+#define RCCHECK(fn) { rcl_ret_t temp_rc = fn; if((temp_rc != RCL_RET_OK)){printf("Failed status on line %d: %d. Aborting.\n",__LINE__,(int)temp_rc);vTaskDelete(NULL);}}
+#define RCSOFTCHECK(fn) { rcl_ret_t temp_rc = fn; if((temp_rc != RCL_RET_OK)){printf("Failed status on line %d: %d. Continuing.\n",__LINE__,(int)temp_rc);}}
+
+//initialization
+rcl_publisher_t publisher;
+std_msgs__msg__String msg;
+int counter = 1;
+
+void timer_callback(rcl_timer_t * timer, int64_t last_call_time)
+{
+	RCLC_UNUSED(last_call_time);
+	if (timer != NULL) {
+		// assign message to publisher
+  		std_msgs__msg__String__init(&msg);
+  		const unsigned int PUB_MSG_CAPACITY = 30;
+  		msg.data.data = malloc(PUB_MSG_CAPACITY);
```

Comment:
This is a memory leak because you are not freeing this resources, please check.

Reply:
Hi,
I have updates the file. Please check once.